### PR TITLE
Added setEnableWordTimeOffsets(true) to Async Recognize for a File

### PR DESCRIPTION
**Pull request by dlaqab**
Added setEnableWordTimeOffsets(true) to Async Recognize for a File
Added setEnableWordTimeOffsets(false) to Sync samples and quick start

### CODE REVIEW

**Comment by dlaqab** [comment]
I am covered by my current employer (Google)

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 1]
```diff
@@ -186,6 +189,11 @@ public static void asyncRecognizeFile(String fileName) throws Exception, IOExcep
       List<SpeechRecognitionAlternative> alternatives = result.getAlternativesList();
       for (SpeechRecognitionAlternative alternative: alternatives) {
         System.out.printf("Transcription: %s%n", alternative.getTranscript());
+        for (WordInfo wordInfo: alternative.getWordsList()) {
+          System.out.println(wordInfo.getWord());
+          System.out.printf("\t%s ns - %s ns\n",
+              wordInfo.getStartTime().getNanos(), wordInfo.getEndTime().getNanos());
```
Comment (gguuss) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Please update the other async sample to correctly show the seconds and nanos calculated to second fractions.

File: speech/cloud-client/src/main/java/com/example/speech/QuickstartSample.java [hunk 2]
```diff
@@ -50,6 +50,7 @@ public static void main(String... args) throws Exception {
         .setEncoding(AudioEncoding.LINEAR16)
         .setSampleRateHertz(16000)
         .setLanguageCode("en-US")
+        .setEnableWordTimeOffsets(false)
```
Comment (gguuss) [inline speech/cloud-client/src/main/java/com/example/speech/QuickstartSample.java]: Isn't this superfluous as false is the default?

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 1 above)
Comment (dlaqab) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: your original code did not have seconds, and I did not change the "pretty print" to show seconds.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 3]
```diff
@@ -186,6 +189,11 @@ public static void asyncRecognizeFile(String fileName) throws Exception, IOExcep
       List<SpeechRecognitionAlternative> alternatives = result.getAlternativesList();
       for (SpeechRecognitionAlternative alternative: alternatives) {
         System.out.printf("Transcription: %s%n", alternative.getTranscript());
+        for (WordInfo wordInfo: alternative.getWordsList()) {
```
Comment (dlaqab) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: Done. I added the test. I also fixed the test failure in AsyncRecognizeGcs test, so all the tests should pass now

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java [hunk 4]
```diff
@@ -185,7 +188,12 @@ public static void asyncRecognizeFile(String fileName) throws Exception, IOExcep
     for (SpeechRecognitionResult result: results) {
       List<SpeechRecognitionAlternative> alternatives = result.getAlternativesList();
       for (SpeechRecognitionAlternative alternative: alternatives) {
-        System.out.printf("Transcription: %s%n", alternative.getTranscript());
+        System.out.printf("Transcription: %s\n",alternative.getTranscript());
```
Comment (gguuss) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: This should be:

```
          System.out.printf("\t%s.%s sec - %s.%s sec\n",
              wordInfo.getStartTime().getSeconds(),
              wordInfo.getStartTime().getNanos() / 100000000,
              wordInfo.getEndTime().getSeconds(),
              wordInfo.getEndTime().getNanos() / 100000000);
        }
```

File: speech/cloud-client/src/test/java/com/example/speech/RecognizeIT.java [hunk 5]
```diff
@@ -84,10 +84,17 @@ public void testAsyncRecognizeGcs() throws Exception {
...
+    String got = bout.toString();
+    assertThat(got).contains("\t0 ns");
+  }
+
+  @Test
+  public void testAsyncWordoffsetGcs() throws Exception {
     Recognize.asyncRecognizeGcs(gcsPath);
     String got = bout.toString();
-    assertThat(got).contains("\t0.0 sec -");
+    assertThat(got).contains("\t0 ns");
```
Comment (gguuss) [inline speech/cloud-client/src/test/java/com/example/speech/RecognizeIT.java]: Revert this change when you correct the print output in `asyncRecognizeFile`

**Comment by gguuss** [comment]
@dlaqab Please update your branch to match master, change your new code to match the updated start seconds / end seconds code as:
```
          System.out.printf("\t%s.%s sec - %s.%s sec\n",
              wordInfo.getStartTime().getSeconds(),
              wordInfo.getStartTime().getNanos() / 100000000,
              wordInfo.getEndTime().getSeconds(),
              wordInfo.getEndTime().getNanos() / 100000000);
        }
```

Revert your change in RecognizeIT to test for actual seconds / fractions of a second.

File: speech/cloud-client/src/main/java/com/example/speech/Recognize.java (hunk 4 above)
Comment (dlaqab) [inline speech/cloud-client/src/main/java/com/example/speech/Recognize.java]: This line shows the all up transcript - lines bellow it are iterating through the words and displaying the start and end time stamps

**Comment by gguuss** [comment]
@lesv /FYI - Looks like the Circle tests are not running because this is on a personal fork of the repo. When I run the tests locally, the following is my output:

> [INFO] Scanning for projects...
> [INFO]
> [INFO] ------------------------------------------------------------------------
> [INFO] Building speech-google-cloud-samples 1.0.5
> [INFO] ------------------------------------------------------------------------
> [INFO]
> [INFO] --- jacoco-maven-plugin:0.7.9:prepare-agent (default) @ speech-google-cloud-samples ---
> [INFO] argLine set to -javaagent:/Users/class/.m2/repository/org/jacoco/org.jacoco.agent/0.7.9/org.jacoco.agent-0.7.9-runtime.jar=destfile=/Users/class/temp/daryush/java-docs-samples/speech/cloud-client/target/jacoco.exec
> [INFO]
> [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ speech-google-cloud-samples ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/main/resources
> [INFO]
> [INFO] --- maven-compiler-plugin:3.6.1:compile (default-compile) @ speech-google-cloud-samples ---
> [INFO] Nothing to compile - all classes are up to date
> [INFO]
> [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ speech-google-cloud-samples ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] skip non existing resourceDirectory /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/test/resources
> [INFO]
> [INFO] --- maven-compiler-plugin:3.6.1:testCompile (default-testCompile) @ speech-google-cloud-samples ---
> [INFO] Nothing to compile - all classes are up to date
> [INFO]
> [INFO] --- maven-surefire-plugin:2.20:test (default-test) @ speech-google-cloud-samples ---
> [INFO]
> [INFO] --- jacoco-maven-plugin:0.7.9:report (report) @ speech-google-cloud-samples ---
> [INFO] Loading execution data file /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/target/jacoco.exec
> [INFO] Analyzed bundle 'speech-google-cloud-samples' with 3 classes
> [INFO]
> [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ speech-google-cloud-samples ---
> [INFO]
> [INFO] --- maven-failsafe-plugin:2.20:integration-test (default) @ speech-google-cloud-samples ---
> [INFO]
> [INFO] -------------------------------------------------------
> [INFO]  T E S T S
> [INFO] -------------------------------------------------------
> [INFO] Running com.example.speech.QuickstartSampleIT
> [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.831 s - in com.example.speech.QuickstartSampleIT
> [INFO] Running com.example.speech.RecognizeIT
> [INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 52.388 s - in com.example.speech.RecognizeIT
> [INFO]
> [INFO] Results:
> [INFO]
> [INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
> [INFO]
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 58.288 s
> [INFO] Finished at: 2017-08-02T16:42:02-07:00
> [INFO] Final Memory: 21M/376M
> [INFO] ------------------------------------------------------------------------

Checkstyle shows:

> [INFO] Scanning for projects...
> [INFO]
> [INFO] ------------------------------------------------------------------------
> [INFO] Building speech-google-cloud-samples 1.0.5
> [INFO] ------------------------------------------------------------------------
> [INFO]
> [INFO] --- maven-checkstyle-plugin:2.17:check (default-cli) @ speech-google-cloud-samples ---
> [INFO] Starting audit...
> [WARN] /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/main/java/com/example/speech/QuickstartSample.java:36:3: Missing a Javadoc comment. [JavadocMethod]
> [WARN] /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/main/java/com/example/speech/Recognize.java:47:3: Missing a Javadoc comment. [JavadocMethod]
> [WARN] /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/main/java/com/example/speech/Recognize.java:157:3: Missing a Javadoc comment. [JavadocMethod]
> [WARN] /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/test/java/com/example/speech/QuickstartSampleIT.java:35: Abbreviation in name 'QuickstartSampleIT' must contain no more than '2' consecutive capital letters. [AbbreviationAsWordInName]
> [WARN] /Users/class/temp/daryush/java-docs-samples/speech/cloud-client/src/test/java/com/example/speech/RecognizeIT.java:35: Abbreviation in name 'RecognizeIT' must contain no more than '2' consecutive capital letters. [AbbreviationAsWordInName]
> Audit done.
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 1.491 s
> [INFO] Finished at: 2017-08-02T16:44:34-07:00
> [INFO] Final Memory: 18M/308M
> [INFO] ------------------------------------------------------------------------

So LGTM.

File: speech/cloud-client/src/main/java/com/example/speech/QuickstartSample.java (hunk 2 above)
Comment (gguuss) [inline speech/cloud-client/src/main/java/com/example/speech/QuickstartSample.java]: I'm pretty sure we don't want to set an optional parameter to its default value in the quickstart example. Thoughts?

**Comment by lesv** [comment]
@gguuss I'll try to fix that tomorrow.

**Comment by gguuss** [comment]
I signed it!

**Comment by remi** [comment]
This can be closed.

**Comment by dlaqab** [comment]
why? I am not sure why this needs to be closed

**Comment by remi** [comment]
1. The default for EnableWordTimeOffsets is false; it is incorrect to add it explicitly.
2. None of the other languages do this, making Java inconsistent.

**Comment by remi** [comment]
The setEnableWordTimeOffsets() call in these samples is extraneous to the task these samples are intended to demonstrate. Closing.

**Comment by remi** [comment]
Note: samples which demonstrate Enable Word Time Offsets were added to java-docs-samples in #787
